python/cudf/cudf/core/column/column.py

```diff
@@ -1020,7 +1020,9 @@ def column_empty(row_count, dtype="object", masked=False):
         data = None
         children = (
             build_column(
-                data=Buffer(cupy.zeros(row_count + 1, dtype="int32")),
+                data=Buffer(
+                    cupy.zeros(row_count + 1, dtype="int32").view("|u1")
+                ),
```

Comment:
This feels pretty cumbersome to me, but would love for others to weigh in here. @shwina @rgsl888prabhu @galipremsagar

Reply:
We should maybe use https://github.com/rapidsai/cudf/blob/branch-0.15/cpp/include/cudf/column/column_factories.hpp#L37 than try to do this ourselves?

Reply:
Especially with more nested column types coming, this will become more cumbersome. I'd rather that the logic for generating empty columns be written once, in the C++

Reply:
I specifically meant in constructing a `Buffer` from a `cupy.array`.

Reply:
I agree with @shwina, we need and API in libcudf to handle the column creation.

But I am not sure about what @kkraus14  is implying regarding cupy.array.

Reply:
It feels cumbersome to me that in order to construct a Buffer from a cupy array we have to view the array as `u1` each time. But that may just be me 😄.

Reply:
I am not sure how else to avoid the situation where a user creates a cupy/numpy array of ints that they want to represent bytes, but accidentally leave it as `int32` or `int64` when constructing the buffer. This might result in scrambled data.

Reply:
Yup I get it, this is fine for now, lets just move forward with this approach.